aws_cloudtrail_rules/aws_iam_user_recon_denied.py

```diff
@@ -0,0 +1,54 @@
+from fnmatch import fnmatch
+from ipaddress import ip_address
+from panther_oss_helpers import check_threshold  # pylint: disable=import-error
+
+# service/event patterns to monitor
+RECON_ACTIONS = {
+    'dynamodb': ['List*', 'Describe*', 'Get*'],
+    'ec2': ['Describe*', 'Get*'],
+    'iam': ['List*', 'Get*'],
+    's3': ['List*', 'Get*'],
+    'rds': ['Describe*', 'List*'],
+}
+THRESH = 5
+THRESH_TTL = 600  # 10 minutes
+
+
+def rule(event):
+    # Filter events
+    if event['errorCode'] != 'AccessDenied':
+        return False
+    if event['userIdentity'].get('type') != 'IAMUser':
+        return False
+
+    # Validate the request came from outside of AWS
+    try:
+        ip_address(event.get('sourceIPAddress'))
+    except ValueError:
+        return False
+
+    # Pattern match this event to the recon actions
+    for event_source, event_patterns in RECON_ACTIONS.items():
+        if event['eventSource'].startswith(event_source):
+            if any([
+                    fnmatch(event['eventName'], event_pattern)
+                    for event_pattern in event_patterns
+            ]):
+                return True
```

Comment:
Using a generator instead of a list is more efficient:

```suggestion
        if event['eventSource'].startswith(event_source) and any(
                fnmatch(event['eventName'], event_pattern)
                for event_pattern in event_patterns
        )):
            return True
```